packages/react/src/tags/Tag.tsx

```diff
@@ -33,7 +43,16 @@ export const Tag = ({ children, onRemove, ...props }: TagProps) => {
 				{children}
 			</Box>
 			{onRemove && (
-				<TagRemoveButton aria-label={`Remove ${children}`} onClick={onRemove} />
+				<TagRemoveButton
+					aria-label={`Remove ${children}`}
+					onClick={(event) => {
+						// We call this first so consumers can manage focus separately in their onRemove
+						if (index && focusOnRemove) {
```

Comment:
Could `index` be 0?

Reply:
Since focusing happens before removing, when you delete the first:
* if it's the only, it will focus itself (i.e. do nothing)
* if it's not, then it will focus itself, but then the removal will move the focus to the new first one (like pagination was doing before)